lib/spack/spack/spec.py

```diff
@@ -1666,6 +1666,9 @@ def _add_flag(self, name, value, propagate):
             self._set_architecture(os=value)
         elif name == "target":
             self._set_architecture(target=value)
+        elif name == "namespace":
+            self.namespace = value
+            assert not propagate
```

Comment:
Is this needed now that we blocked it in the parser?

Reply:
In that we *really* don't want to accept propagation, yes.  In that it'll "never" happen, no, but that's what `assert` is for.  I like your suggestion below, though.